base/reduce.jl

```diff
@@ -634,11 +637,6 @@ function mapreduce_impl(f, op::Union{typeof(max), typeof(min)},
     start = first + 1
     simdstop  = start + chunk_len - 4
     while simdstop <= last - 3
-        # short circuit in case of NaN or missing
-        (v1 == v1) === true || return v1
-        (v2 == v2) === true || return v2
-        (v3 == v3) === true || return v3
-        (v4 == v4) === true || return v4
```

Comment:
This is definitely incorrect if inputs contains both `missing` and `NaN`. (It might return a `NaN`, but we should always return `missing`). Related test is added in the same commit.